### ISSUE TITLE
[PRE REVIEW]: MiSTree: a Python package for constructing and analysing Minimum Spanning Trees

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1690]
**Submitting author:** @knaidoo29 (<a href="http://orcid.org/0000-0002-9182-1802">Krishna Naidoo</a>)
**Repository:** <a href="https://github.com/knaidoo29/mistree" target ="_blank">https://github.com/knaidoo29/mistree</a>
**Version:** v1.1
**Editor:** @xuanxu
**Reviewers:** @melissawm, @zonca

**Author instructions**

Thanks for submitting your paper to JOSS @knaidoo29. **Currently, there isn't an JOSS editor assigned** to your paper.

@knaidoo29 if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```


**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for an editor to be assigned and for them to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01690/joss.01690/10.21105.joss.01690.pdf)

**Comment by labarba** [comment]
👋  @knaidoo29 —  I am the Associate Editor-in-Chief on rotation this week. I'm going to send this back to you for some more work, because the paper does not quite meet [our expectations](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain). The paper, as submitted, is very brief and doesn't give enough of a description of the software or its need and uses in research. We don't expect a full article, but we recommend 250–1000 words (your Summary is just under that). These items need to be included:

- the summary describing high-level functionality and purpose of the software for a diverse, non-specialist audience;
- a clear statement of need that illustrates the research application of the software (including references to other software meeting similar needs).

TL;DR: the article needs to be improved with more description of how the software works, some examples, and use cases.

**Comment by labarba** [comment]
@xuanxu — Could you handle this submission as editor?

**Comment by knaidoo29** [comment]
Hi @labarba, thank you for the comments. I am currently working on a significant revision/rewrite of the paper and was wondering (1) what the timescale for edits could be (I am planning to have this ready within a week) and (2) what the process was for resubmitting (would I just need to push my new version of the paper to the software's repository?).

**Comment by labarba** [comment]
You can just push your changes to the paper, and let us know here when ready. I'll set a reminder for you.

**Comment by labarba** [comment]
@whedon remind @knaidoo29 in one week

**Comment by whedon** [comment]
Sorry, I can't set reminders on PRE-REVIEW issues.

**Comment by labarba** [comment]
Oops. I just learned that our bot's reminding powers don't work here. We'll just wait for your ping!

**Comment by xuanxu** [comment]
@whedon add @xuanxu as editor

**Comment by xuanxu** [comment]
@whedon assign @xuanxu as editor

**Comment by whedon** [comment]
OK, the editor is @xuanxu

**Comment by knaidoo29** [comment]
Hi @labarba and @xuanxu just to let you know I've updated the paper and pushed it to the repository. I rewrote it and have added some basic examples. 

Note, that I've added a new version of the code (i.e. v1.1) which includes some new functions for binning and plotting the Minimum Spanning Tree statistics. I'm not familiar with how you will review the code but I just wanted to highlight this as I believe I placed v1.0 as the version to be tested.

I look forward to your comments.

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01690/joss.01690/10.21105.joss.01690.pdf)

**Comment by xuanxu** [comment]
@knaidoo29 great, thanks! 
We'll start the review once the reviewers are assigned.

**Comment by knaidoo29** [comment]
Hi @xuanxu, I just had a look at the proofs and realised my figures are not there. This was because I had my gitignore set up to ignore png files. The images are now on the repository so if the pdf is regenerated it should appear on it now. Thanks (p.s can I regenerate the PDF or do I have to wait for you to do it?)

**Comment by labarba** [comment]
You can also run `@whedon generate pdf` (as a single line in a separate comment) to compile the proof at any time.

**Comment by knaidoo29** [comment]
Great, will try it now

**Comment by knaidoo29** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01690/joss.01690/10.21105.joss.01690.pdf)

**Comment by xuanxu** [comment]
:wave: @zonca: would you be willing to review this paper? I think your knowledge of Python and algorithms applied to cosmology makes you a great fit for this submission.

**Comment by xuanxu** [comment]
:wave: @melissawm: would you be willing to review this paper? I think your knowledge of math and Python+Fortran makes you a great fit for this submission.

**Comment by melissawm** [comment]
Sure, count me in!

**Comment by xuanxu** [comment]
@melissawm: great, thanks!
I'm going to assign you now and then I'll start the review issue once I find the second reviewer, stay tuned!

**Comment by xuanxu** [comment]
@whedon assign @melissawm as reviewer

**Comment by whedon** [comment]
OK, the reviewer is @melissawm

**Comment by zonca** [comment]
@xuanxu thanks, yes, I can review this

**Comment by xuanxu** [comment]
@zonca great, thanks! Let's get this started.

**Comment by xuanxu** [comment]
@whedon add @zonca as reviewer

**Comment by whedon** [comment]
OK, @zonca is now a reviewer

**Comment by xuanxu** [comment]
@whedon start review

**Comment by xuanxu** [comment]
@melissawm, @zonca : please head over to #1721 where the review will take place.
